resources/assets/coffee/react/beatmaps/search-filter.coffee

```diff
@@ -50,10 +56,22 @@ class @SearchFilter extends React.Component
       else
         @setState { selected: [ i ] }, @triggerUpdate
 
+  clickReject: (e) ->
+    e.preventDefault()
+
+  componentWillReceiveProps: (props) ->
+    if @state.selected != props.selected
```

Comment:
why not move the whole state to parent instead of dealing with synchronizing this thing here?